restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/VoiceInterpreter.java

```diff
@@ -1101,7 +1101,13 @@ private void onCallStateChanged(Object message, ActorRef sender) throws Transiti
                         if (dialBranches != null && dialBranches.contains(sender)) {
                             removeDialBranch(message, sender);
                         }
-                        checkDialBranch(message, sender, attribute);
+                        if (dialBranches != null && !dialBranches.isEmpty()){
```

Comment:
I see, thanks @gvagenas ! But i understood from a previous conversation on slack with you and @deruelle  that if **all** or **the only** dial branch are/is busy, the interpreter should not proceed with the RCML and disconnecting the call instead. Please check [this message](https://telestax.slack.com/archives/restcomm-connect/p1482484549001229). Because of that, i assumed that `checkDialBranch` should be performed only if `dialBranches` is not empty, otherwise the call should be finished.

But, if i got it wrong and it has to be the way you mentioned now by steps 1, 2 and 3, i suppose we'll face that exception about an invalid transition from `forking` state to `checkingCache` state, since the flow goes from a Dial verb receiving busy to a Play verb.

I wait for your consideration on that to continue. Thanks! :)